arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/section/halt/SelfdestructSection.java

```diff
@@ -293,13 +284,24 @@ public void resolveAtEndTransaction(
                   DomSubStampsSubFragment.selfdestructDomSubStamps(hub, hubStamp));
 
       this.addFragment(accountWipingFragment);
+      this.addFragment(finalUnexceptionalContextFragment);
+
+      hub.defers().scheduleForAfterTransactionFinalization(this);
     } else {
-      selfdestructScenarioFragment.setScenario(
-          SelfdestructScenarioFragment.SelfdestructScenario
-              .SELFDESTRUCT_WONT_REVERT_ALREADY_MARKED);
+      selfdestructScenarioFragment.setScenario(SELFDESTRUCT_WONT_REVERT_ALREADY_MARKED);
+      this.addFragment(finalUnexceptionalContextFragment);
     }
+  }
+
+  @Override
+  public void resolveAfterTransactionFinalization(Hub hub, WorldView state) {
+
+    hub.transients()
+        .conflation()
+        .deploymentInfo()
+        .deploymentUpdateForSuccessfulSelfDestruct(selfdestructor.address());
 
-    this.addFragment(finalUnexceptionalContextFragment);
+    accountWipingNew.wipe(hub.transients().conflation().deploymentInfo());
```

Comment:
This is the raison d'être of the `AfterTransactionFinalization` defer registry. It is so that we don't raise the deployment number of `SELFDESTRUCT`'ing accounts prior to executing the `TX_FINL` phase. This messes things up in case the `COINBASE` is among the `SELFDESTRUCT`'ing accounts.